allennlp/tests/data/token_indexers/character_token_indexer_test.py

```diff
@@ -61,3 +61,88 @@ def test_start_and_end_tokens(self):
+                                    [15]]}
+        key_padding_lengths = "num_token_characters"
+        value_padding_lengths = 0
+        for token in indices["char"]:
+            item = indexer.get_padding_lengths(token)
+            key, value = item.keys(), item.values()
+            assert set([key_padding_lengths]) == set(key)
+            value_padding_lengths = max(value_padding_lengths, max(value))
+        assert value_padding_lengths == 8
+        padded = indexer.pad_token_sequence(indices,
+                                            {"char": len(indices["char"])},
+                                            {key_padding_lengths: value_padding_lengths})
+        assert padded == {"char": [[2, 3, 3, 4, 5, 6, 7, 8],
+                                   [9, 10, 0, 0, 0, 0, 0, 0],
+                                   [11, 12, 4, 10, 13, 14, 4, 0],
+                                   [15, 0, 0, 0, 0, 0, 0, 0]]}
+
+        indexer = TokenCharactersIndexer("characters", min_padding_length=3)
+        indices = indexer.tokens_to_indices(tokens, vocab, "char")
+        assert indices == {"char": [[2, 3, 3, 4, 5, 6, 7, 8],
+                                    [9, 10],
+                                    [11, 12, 4, 10, 13, 14, 4],
+                                    [15]]}
+        key_padding_lengths = "num_token_characters"
+        value_padding_lengths = 0
+        for token in indices["char"]:
+            item = indexer.get_padding_lengths(token)
+            key, value = item.keys(), item.values()
+            assert set([key_padding_lengths]) == set(key)
+            value_padding_lengths = max(value_padding_lengths, max(value))
+        assert value_padding_lengths == 8
+        padded = indexer.pad_token_sequence(indices,
+                                            {"char": len(indices["char"])},
+                                            {key_padding_lengths: value_padding_lengths})
+        assert padded == {"char": [[2, 3, 3, 4, 5, 6, 7, 8],
+                                   [9, 10, 0, 0, 0, 0, 0, 0],
+                                   [11, 12, 4, 10, 13, 14, 4, 0],
+                                   [15, 0, 0, 0, 0, 0, 0, 0]]}
+
+        indexer = TokenCharactersIndexer("characters", min_padding_length=10)
+        indices = indexer.tokens_to_indices(tokens, vocab, "char")
+        assert indices == {"char": [[2, 3, 3, 4, 5, 6, 7, 8],
+                                    [9, 10],
+                                    [11, 12, 4, 10, 13, 14, 4],
+                                    [15]]}
+        key_padding_lengths = "num_token_characters"
+        value_padding_lengths = 0
+        for token in indices["char"]:
+            item = indexer.get_padding_lengths(token)
+            key, value = item.keys(), item.values()
+            assert set([key_padding_lengths]) == set(key)
+            value_padding_lengths = max(value_padding_lengths, max(value))
+        assert value_padding_lengths == 10
```

Comment:
There's no need for lines 130 through 141; you can just go straight to computing the padding, and just have the assert that's on line 145.  The rest is all implicitly tested with that one assert.